component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioDbManager.java

```diff
@@ -0,0 +1,193 @@
+package org.fidoalliance.fdo.sample;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import javax.sql.DataSource;
+
+/**
+ * AIO Database Manager.
+ */
+public class AioDbManager {
+
+  /**
+   * Creates SQL tables for AIO.
+   *
+   * @param ds A SQL datasource
+   */
+  public void createTables(DataSource ds) {
+    try (Connection conn = ds.getConnection();
+        Statement stmt = conn.createStatement()) {
+
+      String sql =
+          "CREATE TABLE IF NOT EXISTS "
+              + "TO2_CONFIG ("
+              + "ID INT NOT NULL DEFAULT 1, "
+              + "INITIALIZED BOOLEAN NOT NULL DEFAULT FALSE, "
+              + "RV_BLOB VARCHAR(4096) NOT NULL DEFAULT 'http://localhost:8080', "
+              + "UNIQUE (ID)"
+              + ");";
+
+      stmt.executeUpdate(sql);
+
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * loads the RvBlob String.
+   * @param ds A SQL datasource
+   * @return The string encoded redirect blob.
+   */
+  public String loadRvBlob(DataSource ds) {
+    String result = "";
+    String sql = "SELECT RV_BLOB FROM TO2_CONFIG WHERE ID = 1;";
+
+    try (Connection conn = ds.getConnection();
+        Statement stmt = conn.createStatement()) {
+
+      try (ResultSet rs = stmt.executeQuery(sql)) {
+        if (rs.next()) {
+          result = rs.getString(1);
+        }
+
+      } catch (SQLException e) {
+        throw new RuntimeException(e);
+      }
+
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+    return result;
+  }
+
+  /**
+   * Stores a signed redirect blob to the database.
+   *
+   * @param guid        The quid of the device.
+   * @param signedBlob  The signed redirect blob.
+   * @param fingerPrint The fingerprint of the owner public key.
+   * @param deviceX509  The device certificate.
+   * @param ds          A SQL datasource.
+   */
+  public void storeRedirectBlob(String guid, byte[] signedBlob,
+      String fingerPrint,
+      byte[] deviceX509,
+      DataSource ds) {
+
+    String sql = ""
+        + "MERGE INTO RV_REDIRECTS  "
+        + "KEY (GUID) "
+        + "VALUES (?,?,?,?,?,?,?);";
+
+    try (Connection conn = ds.getConnection();
+        PreparedStatement pstmt = conn.prepareStatement(sql)) {
+
+      pstmt.setString(1, guid);
+      pstmt.setBytes(2, signedBlob);
+      pstmt.setString(3, fingerPrint);
+      pstmt.setBytes(4, deviceX509);
+      pstmt.setInt(5, 315400000);
```

Comment:
It would be good to define 315400000 as a constant (explaining how we reach this number) and use here.